lib/DDG/Spice/TLDs.pm

```diff
@@ -0,0 +1,79 @@
+package DDG::Spice::TLDs;
+#ABSTRACT Returns the country or purpose of a TLD (top level domain). 
+
+use DDG::Spice;
+use Data::Validate::Domain qw(is_domain);
+use List::Util qw(pairmap);
+use URI::Escape;
+use DDG::Util::SpiceConstants;
+
+
+spice proxy_cache_valid => '418 1d';
+
+my $namecheap_endpoint = 'http://api.namecheap.com/xml.response';
+
+# environment variables:
+# DDG_SPICE_NAMECHEAP_USERNAME : Namecheap API username
+# DDG_SPICE_NAMECHEAP_APIKEY   : Namecheap API key
+# DDG_SPICE_NAMECHEAP_IP_ADDR  : Allowed end-User IP address
+my @query_list = (
+    ApiUser    => 'javathunderman', # Username required to access the API
+    ApiKey     => 'bac485f35c51466caed5ff499e585528',   # Password required used to access the API
+    UserName   => 'javathunderman', # same as ApiUser
+    Command    => 'namecheap.domains.gettldlist',               # API method
+    ClientIp   => '0.0.0.0',  # End-user IP address
+    DomainList => '$1',                                    # argument to method
```

Comment:
I wasn't entirely sure about that. The documentation they provide treats it as if it's spitting out all the domain names at the same time, but the code examples they give are a bit different...If that's the case then I guess this could be implemented as a CheatSheet rather than this...